webapp/src/components/ParcelTags/ParcelTags.js

```diff
@@ -65,8 +70,35 @@ export default class ParcelTags extends React.PureComponent {
     )
   }
 
+  getTags() {
+    if (this.isParcel()) {
+      return this.props.parcel.tags
+    }
+
+    const { parcels } = this.props.estate
+    const proximities = parcels
+      .map(parcel => parcel.tags.proximity)
+      .filter(prox => prox)
+    const proximity = proximities.reduce((finalProximity, actualProximity) => {
+      Object.keys(actualProximity).forEach(key => {
+        if (!finalProximity[key]) {
+          finalProximity[key] = actualProximity[key]
+          return
+        }
+
+        if (finalProximity[key].distance > actualProximity[key].distance) {
+          finalProximity[key] = actualProximity[key]
+          return
+        }
+      })
+      return finalProximity
+    }, {})
+
+    return { proximity }
+  }
+
```

Comment:
_Not imperative_ but maybe this method could look like this?

```
getTags() {
  if (isParcel()) {
     return this.getParcelTags()
  } else {
     return this.getStateTags()
  }
```

just to prepare the reader for each method definition.
Again I'm not 100% sure it's the way to go

Reply:
I'm not sure either, but done 9d2c76c :)